### ISSUE TITLE
Couldn't access conflicts view

### ISSUE THREAD

**Issue by PeteS4** [davideshay/groceries#108]
Hi,

I'm trying to setup this app on my raspi using podman. I have the couchdb-3-container up and running, but when I try to start the groceries-server, after some setup the container gets stuck at an error "Couldn't access conflicts view: Error: missing_named_view". But the log even shows 'Checking/Creating conflicts view...' and then 'View created/ updated'. Any idea what's going on here?

<details>
<summary>Here is the full log output (minus the 21498014 initial products :-) )</summary>

```
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Starting up auth server for couchdb...
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: App Version:  1.1.1
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Database Schema Version: 6
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Database URL:  http://groceries-db:5984
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Internal Database URL:  http://groceries-db:5984
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Using database: groceries
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Refresh token expires in  30d(2592000 seconds)
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: STATUS: Access token expires in  12h(43200 seconds)
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: STATUS: User Account Creation is:  ENABLED
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Database existence checked/created.
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Checking Database security settings
Oct 02 12:42:05 localhost groceries[16181]: [10:42:05] INFO: Security database settings need to be created/updated...
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] ERROR: STATUS: Database security roles added
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: JWT Key verified to access database
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: UUID created in DB:  f76eb196-303f-45ea-b4cb-f5b25c485c9a
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Current Schema Version: 0  Target Version: 6
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updating schema to rev. 2: Changes for 'stocked at' indicator on item/list.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Upgrading schema to support stocked at indicators.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found items to update : 0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Finished schema updates, updating database to : 2
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updated schema target version successfully.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updating schema to rev. 3: Changes for restructuring/listgroups 
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Upgrading schema to support listGroups. Most data will be lost in this upgrade.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Finished deleting lists, items, and categories.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Creating default listgroups for all users
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found users to create listgroups:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Finished schema updates, updating database to : 3
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updated schema target version successfully.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updating schema to rev. 4: Make Categories/UOMs listgroup specific 
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Upgrading schema to link categories,UOMs, and Recipes to list Groups.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Loading up all current categories
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found categories to process:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found items to process:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found users to create listgroups:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found settings to create color specific categories:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Removing default flag from list groups: 0  found to process
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Adding recipe list group for each user: 0  users found to process
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Adding recipe list group for each user: 0  users found to process
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Retrieving updated categories after changes...
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found list groups to create color specific categories:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Generating user-specific colors for every category
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: User Color settings all created/updated 
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: About to update units of measure with listgroup data. Found:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: About to create units of measure for Recipes. Found recipes:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: UOMs by item classified/created
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Deleting color field from categories
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Finished schema updates, updating database to : 4
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updated schema target version successfully.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updating schema to rev 5: Make Images for items listgroup specific 
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Adding list group field to all images...
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Finding all items that have an image...
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found images to process:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Finished schema updates, updating database to : 5
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updated schema target version successfully.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updating schema to rev 6: Added impacted users to conflictlog records 
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Restructuring conflict log records to add impacted users for replication log.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Found conflict logs to process:  0
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Conflict Log records updated with impacted users.
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Finished schema updates, updating database to : 6
Oct 02 12:42:06 localhost groceries[16181]: [10:42:06] INFO: Updated schema target version successfully.
Oct 02 12:42:07 localhost groceries[16181]: [10:42:07] INFO: Fixing Alexa Default list groups / settings
Oct 02 12:42:07 localhost groceries[16181]: [10:42:07] INFO: Alexa default records fixed
Oct 02 12:42:07 localhost groceries[16181]: [10:42:07] INFO: Current UOM Content Version: 0  Target Version: 5
Oct 02 12:42:07 localhost groceries[16181]: [10:42:07] INFO: Creating UOM Content...
Oct 02 12:42:07 localhost groceries[16181]: [10:42:07] INFO: Adding uom  EA   Each
...
Oct 02 12:42:08 localhost groceries[16181]: [10:42:08] INFO: Adding uom  Y3   Teaspoon
Oct 02 12:42:08 localhost groceries[16181]: [10:42:08] INFO: Finished adding units of measure, updating to UOM Content Version: 5
Oct 02 12:42:08 localhost groceries[16181]: [10:42:08] INFO: Updated UOM Target Version successfully.
Oct 02 12:42:08 localhost groceries[16181]: [10:42:08] INFO: Current Category Content Version: 0  Target Version: 2
Oct 02 12:42:08 localhost groceries[16181]: [10:42:08] INFO: Creating category Content...
Oct 02 12:42:08 localhost groceries[16181]: [10:42:08] INFO: Adding category  Bakery
...
Oct 02 12:42:09 localhost groceries[16181]: [10:42:09] INFO: Adding category  Coffee/Tea
Oct 02 12:42:09 localhost groceries[16181]: [10:42:09] INFO: Finished adding categories, updating to category Version: 2
Oct 02 12:42:09 localhost groceries[16181]: [10:42:09] INFO: Updated Categories Target Version successfully.
Oct 02 12:42:09 localhost groceries[16181]: [10:42:09] INFO: Current Global Item Content Version: 0  Target Version: 2
Oct 02 12:42:09 localhost groceries[16181]: [10:42:09] INFO: Creating Global Item Content...
Oct 02 12:42:09 localhost groceries[16181]: [10:42:09] INFO: Adding global item  Bananas
...
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: Adding global item  Clementines
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: Finished adding global Items, updating to Global Item Version: 2
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: Updated Global Item Target Version successfully.
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: Checking/Creating conflicts view...
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: View created/ updated
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: Creating Standard Indexes for find command
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: Checking/Creating index  stdType
Oct 02 12:42:16 localhost groceries[16181]: [10:42:16] INFO: Could not retrieve index stdType... Creating...
Oct 02 12:42:17 localhost groceries[16181]: [10:42:17] INFO: Adding type label to index...
Oct 02 12:42:17 localhost groceries[16181]: [10:42:17] INFO: Checking/Creating index  stdTypeName
Oct 02 12:42:17 localhost groceries[16181]: [10:42:17] INFO: Could not retrieve index stdTypeName... Creating...
Oct 02 12:42:17 localhost groceries[16181]: [10:42:17] INFO: Adding type label to index...
Oct 02 12:42:18 localhost groceries[16181]: [10:42:18] INFO: Checking/Creating index  stdTypeUsername
Oct 02 12:42:18 localhost groceries[16181]: [10:42:18] INFO: Could not retrieve index stdTypeUsername... Creating...
Oct 02 12:42:18 localhost groceries[16181]: [10:42:18] INFO: Adding type label to index...
Oct 02 12:42:18 localhost groceries[16181]: [10:42:18] INFO: Checking/Creating index  stdTypeListGroupID
Oct 02 12:42:18 localhost groceries[16181]: [10:42:18] INFO: Could not retrieve index stdTypeListGroupID... Creating...
Oct 02 12:42:18 localhost groceries[16181]: [10:42:18] INFO: Adding type label to index...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Checking/Creating index  stdTypeLists
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Could not retrieve index stdTypeLists... Creating...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Adding type label to index...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Checking/Creating index  stdFriend
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Could not retrieve index stdFriend... Creating...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Adding type label to index...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Checking/Creating index  stdConflict
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Could not retrieve index stdConflict... Creating...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Adding type label to index...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Checking/Creating index  stdTypeOwnerDefault
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Could not retrieve index stdTypeOwnerDefault... Creating...
Oct 02 12:42:19 localhost groceries[16181]: [10:42:19] INFO: Adding type label to index...
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] INFO: Replication Filter does not exist... Need to create...
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] INFO: Replication filter created/updated successfully.
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] INFO: Conflict resolution scheduled every  15  minutes.
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] INFO: JWT expiry scheduled every  60  minutes.
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] INFO: Checking for expired JWTs
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] INFO: Startup process completed
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] INFO: Finished checking for expired JWTs
Oct 02 12:42:20 localhost groceries[16181]: [10:42:20] ERROR: Couldn't access conflicts view: Error: missing_named_view
Oct 02 12:42:20 localhost groceries[16181]:     at responseHandler (/usr/app/node_modules/nano/lib/nano.js:206:20)
Oct 02 12:42:20 localhost groceries[16181]:     at /usr/app/node_modules/nano/lib/nano.js:456:13
Oct 02 12:42:20 localhost groceries[16181]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
Oct 02 12:42:20 localhost groceries[16181]:   scope: 'couch',
Oct 02 12:42:20 localhost groceries[16181]:   statusCode: 404,
Oct 02 12:42:20 localhost groceries[16181]:   request: {
Oct 02 12:42:20 localhost groceries[16181]:     method: 'get',
Oct 02 12:42:20 localhost groceries[16181]:     headers: {
Oct 02 12:42:20 localhost groceries[16181]:       'content-type': 'application/json',
Oct 02 12:42:20 localhost groceries[16181]:       accept: 'application/json',
Oct 02 12:42:20 localhost groceries[16181]:       'user-agent': 'nano/10.1.2 (Node.js v20.6.0)',
Oct 02 12:42:20 localhost groceries[16181]:       'Accept-Encoding': 'deflate, gzip',
Oct 02 12:42:20 localhost groceries[16181]:       Authorization: 'Basic xxx=='
Oct 02 12:42:20 localhost groceries[16181]:     },
Oct 02 12:42:20 localhost groceries[16181]:     qsStringifyOptions: { arrayFormat: 'repeat' },
Oct 02 12:42:20 localhost groceries[16181]:     url: 'http://groceries-db:5984/groceries/_design/_conflicts_only_view_id/_view/conflicts_view',
Oct 02 12:42:20 localhost groceries[16181]:     params: undefined,
Oct 02 12:42:20 localhost groceries[16181]:     paramsSerializer: { serialize: [Function: serialize] },
Oct 02 12:42:20 localhost groceries[16181]:     data: undefined,
Oct 02 12:42:20 localhost groceries[16181]:     maxRedirects: 0,
Oct 02 12:42:20 localhost groceries[16181]:     httpAgent: Agent {
Oct 02 12:42:20 localhost groceries[16181]:       _events: [Object: null prototype],
Oct 02 12:42:20 localhost groceries[16181]:       _eventsCount: 2,
Oct 02 12:42:20 localhost groceries[16181]:       _maxListeners: undefined,
Oct 02 12:42:20 localhost groceries[16181]:       defaultPort: 80,
Oct 02 12:42:20 localhost groceries[16181]:       protocol: 'http:',
Oct 02 12:42:20 localhost groceries[16181]:       options: [Object: null prototype],
Oct 02 12:42:20 localhost groceries[16181]:       requests: [Object: null prototype] {},
Oct 02 12:42:20 localhost groceries[16181]:       sockets: [Object: null prototype] {},
Oct 02 12:42:20 localhost groceries[16181]:       freeSockets: [Object: null prototype],
Oct 02 12:42:20 localhost groceries[16181]:       keepAliveMsecs: 30000,
Oct 02 12:42:20 localhost groceries[16181]:       keepAlive: true,
Oct 02 12:42:20 localhost groceries[16181]:       maxSockets: 50,
Oct 02 12:42:20 localhost groceries[16181]:       maxFreeSockets: 256,
Oct 02 12:42:20 localhost groceries[16181]:       scheduling: 'lifo',
Oct 02 12:42:20 localhost groceries[16181]:       maxTotalSockets: Infinity,
Oct 02 12:42:20 localhost groceries[16181]:       totalSocketCount: 2,
Oct 02 12:42:20 localhost groceries[16181]:       [Symbol(kCapture)]: false
Oct 02 12:42:20 localhost groceries[16181]:     },
Oct 02 12:42:20 localhost groceries[16181]:     httpsAgent: Agent {
Oct 02 12:42:20 localhost groceries[16181]:       _events: [Object: null prototype],
Oct 02 12:42:20 localhost groceries[16181]:       _eventsCount: 2,
Oct 02 12:42:20 localhost groceries[16181]:       _maxListeners: undefined,
Oct 02 12:42:20 localhost groceries[16181]:       defaultPort: 443,
Oct 02 12:42:20 localhost groceries[16181]:       protocol: 'https:',
Oct 02 12:42:20 localhost groceries[16181]:       options: [Object: null prototype],
Oct 02 12:42:20 localhost groceries[16181]:       requests: [Object: null prototype] {},
Oct 02 12:42:20 localhost groceries[16181]:       sockets: [Object: null prototype] {},
Oct 02 12:42:20 localhost groceries[16181]:       freeSockets: [Object: null prototype] {},
Oct 02 12:42:20 localhost groceries[16181]:       keepAliveMsecs: 30000,
Oct 02 12:42:20 localhost groceries[16181]:       keepAlive: true,
Oct 02 12:42:20 localhost groceries[16181]:       maxSockets: 50,
Oct 02 12:42:20 localhost groceries[16181]:       maxFreeSockets: 256,
Oct 02 12:42:20 localhost groceries[16181]:       scheduling: 'lifo',
Oct 02 12:42:20 localhost groceries[16181]:       maxTotalSockets: Infinity,
Oct 02 12:42:20 localhost groceries[16181]:       totalSocketCount: 0,
Oct 02 12:42:20 localhost groceries[16181]:       maxCachedSessions: 100,
Oct 02 12:42:20 localhost groceries[16181]:       _sessionCache: [Object],
Oct 02 12:42:20 localhost groceries[16181]:       [Symbol(kCapture)]: false
Oct 02 12:42:20 localhost groceries[16181]:     }
Oct 02 12:42:20 localhost groceries[16181]:   },
Oct 02 12:42:20 localhost groceries[16181]:   headers: {
Oct 02 12:42:20 localhost groceries[16181]:     uri: 'http://groceries-db:5984/groceries/_design/_conflicts_only_view_id/_view/conflicts_view',
Oct 02 12:42:20 localhost groceries[16181]:     statusCode: 404,
Oct 02 12:42:20 localhost groceries[16181]:     'cache-control': 'must-revalidate',
Oct 02 12:42:20 localhost groceries[16181]:     'content-type': 'application/json',
Oct 02 12:42:20 localhost groceries[16181]:     date: 'Mon, 02 Oct 2023 10:42:19 GMT',
Oct 02 12:42:20 localhost groceries[16181]:     'x-couch-request-id': '20528e12d6',
Oct 02 12:42:20 localhost groceries[16181]:     'x-couchdb-body-time': '0'
Oct 02 12:42:20 localhost groceries[16181]:   },
Oct 02 12:42:20 localhost groceries[16181]:   errid: 'non_200',
Oct 02 12:42:20 localhost groceries[16181]:   description: 'missing_named_view',
Oct 02 12:42:20 localhost groceries[16181]:   error: 'not_found',
Oct 02 12:42:20 localhost groceries[16181]:   reason: 'missing_named_view'
Oct 02 12:42:20 localhost groceries[16181]: }
```

</details>

**Comment by davideshay** [comment]
Thanks for the feedback. I've now corrected this in release 1.1.2. Two things were found:

- On new installs, the database name indicated as an environment variable should be created automatically. So please delete the. database and try again.
- There was an error when the conflicts view was being created.

Please try again after deleting the DB and let me know how it works.

**Comment by PeteS4** [comment]
Thanks for the quick response, now it works! :+1:
